Content.Server/Standing/LayingDownSystem.cs

```diff
@@ -0,0 +1,94 @@
+    {
+        CommandBinds.Builder
+            .Bind(ContentKeyFunctions.ToggleStanding, InputCmdHandler.FromDelegate(ToggleStanding, handle: false, outsidePrediction: false))
+            .Register<LayingDownSystem>();
+
+        SubscribeLocalEvent<LayingDownComponent, StoodEvent>(DoRefreshMovementSpeed);
+        SubscribeLocalEvent<LayingDownComponent, DownedEvent>(DoRefreshMovementSpeed);
+        SubscribeLocalEvent<LayingDownComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovementSpeed);
+    }
+
+    public override void Shutdown()
+    {
+        base.Shutdown();
+
+        CommandBinds.Unregister<LayingDownSystem>();
+    }
+
+    private void DoRefreshMovementSpeed(EntityUid uid, LayingDownComponent component, object args)
+    {
+        _movement.RefreshMovementSpeedModifiers(uid);
+    }
+
+    private void OnRefreshMovementSpeed(EntityUid uid, LayingDownComponent component, RefreshMovementSpeedModifiersEvent args)
+    {
+        if (TryComp<StandingStateComponent>(uid, out var standingState) && standingState.Standing)
+            return;
+
+        args.ModifySpeed(component.DownedSpeedMultiplier, component.DownedSpeedMultiplier);
+    }
+
+    private void ToggleStanding(ICommonSession? session)
+    {
+        if (session is not { } playerSession)
+            return;
+
+        if ((playerSession.AttachedEntity is not { Valid: true } uid || !Exists(uid)))
+            return;
+
+        if (!TryComp<StandingStateComponent>(uid, out var standingState) || !TryComp<LayingDownComponent>(uid, out var layingDown))
+            return;
+
+        var success = ToggleStandingImpl(uid, standingState, layingDown, out var popupBranch);
+
+        // If successful, show popup to self and others. Otherwise, only to self.
+        _popups.PopupEntity(Loc.GetString($"laying-comp-{popupBranch}-self", ("entity", uid)), uid, uid);
+
+        if (success)
+        {
+            _popups.PopupEntity(Loc.GetString($"laying-comp-{popupBranch}-other", ("entity", uid)), uid, Filter.PvsExcept(uid), true);
+
+            layingDown.CooldownUntil = _timing.CurTime + layingDown.Cooldown;
+        }
+    }
+
+    private bool ToggleStandingImpl(EntityUid uid, StandingStateComponent standingState, LayingDownComponent layingDown, out string popupBranch)
+    {
+        var success = layingDown.CooldownUntil <= _timing.CurTime;
+
+        // Note: &= leads to the right-hand side being evaluated regardless of what's on left-hand, so we use normal assignment here instead.
+        if (_standing.IsDown(uid, standingState))
+        {
+            success = success && _standing.Stand(uid, standingState, force: false);
+            popupBranch = success ? "stand-success" : "stand-fail";
+        }
+        else
+        {
+            success = success && _standing.Down(uid, standingState: standingState, playSound: true, dropHeldItems: false);
+            popupBranch = success ? "lay-success" : "lay-fail";
+        }
```

Comment:
Why are you modifying `popupBranch` after all usages?

Reply:
because there's 4 different branches depending on whether the entity is standing and whether the attempt to toggle state is successful.

Reply:
Oh, I see it's an `out`.